craft_application/application.py

```diff
@@ -629,18 +595,18 @@ def _run_inner(self) -> int:
             platform = platform.split(",", maxsplit=1)[0]
         if build_for and "," in build_for:
             build_for = build_for.split(",", maxsplit=1)[0]
+        if command.needs_project(dispatcher.parsed_args()):
+            project_service = self.services.get("project")
+            # This branch always runs, except during testing.
+            if not project_service.is_rendered:
+                project_service.render_once()
```

Comment:
what about the `platform` and `build_for` parameters? How do they end up in the service in this specific code path?

Reply:
Good catch, thanks!